tests/test_extensions/test_sklearn_extension/test_sklearn_extension.py

```diff
@@ -1254,101 +1273,259 @@ def test_seed_model_raises(self):
             with self.assertRaises(ValueError):
                 self.extension.seed_model(model=clf, seed=42)
 
-    def test_run_model_on_fold(self):
-        task = openml.tasks.get_task(7)
-        num_instances = 320
+    def test_run_model_on_fold_classification_1(self):
+        task = openml.tasks.get_task(1)
         num_folds = 1
         num_repeats = 1
 
-        clf = sklearn.linear_model.SGDClassifier(loss='log', random_state=1)
+        X, y = task.get_X_and_y()
+        train_indices, test_indices = task.get_train_test_split_indices(
+            repeat=0, fold=0, sample=0)
+        X_train = X[train_indices]
+        y_train = y[train_indices]
+        X_test = X[test_indices]
+        y_test = y[test_indices]
+
+        pipeline = sklearn.pipeline.Pipeline(steps=[
+            ('imp', sklearn.preprocessing.Imputer()),
+            ('clf', sklearn.tree.DecisionTreeClassifier()),
+        ])
         # TODO add some mocking here to actually test the innards of this function, too!
         res = self.extension._run_model_on_fold(
-            clf, task, 0, 0, 0,
-            add_local_measures=True)
+            model=pipeline,
+            task=task,
+            fold_no=0,
+            rep_no=0,
+            X_train=X_train,
+            y_train=y_train,
+            X_test=X_test,
+            classes=task.class_labels,
+        )
+
+        y_hat, y_hat_proba, user_defined_measures, trace = res
 
-        arff_datacontent, arff_tracecontent, user_defined_measures, model = res
         # predictions
-        self.assertIsInstance(arff_datacontent, list)
+        self.assertIsInstance(y_hat, np.ndarray)
+        self.assertEqual(y_hat.shape, y_test.shape)
+        self.assertIsInstance(y_hat_proba, np.ndarray)
+        self.assertEqual(y_hat_proba.shape, (y_test.shape[0], 6))
+        np.testing.assert_array_almost_equal(np.sum(y_hat_proba, axis=1), np.ones(y_test.shape))
+        # The class '4' (at index 3) is not present in the training data. We check that the
+        # predicted probabilities for that class are zero!
+        np.testing.assert_array_almost_equal(y_hat_proba[:, 3], np.zeros(y_test.shape))
+        for i in (0, 1, 2, 4, 5):
+            self.assertTrue(np.any(y_hat_proba[:, i] != np.zeros(y_test.shape)))
+
+        # check user defined measures
+        fold_evaluations = collections.defaultdict(lambda: collections.defaultdict(dict))
+        for measure in user_defined_measures:
+            fold_evaluations[measure][0][0] = user_defined_measures[measure]
+
         # trace. SGD does not produce any
-        self.assertIsInstance(arff_tracecontent, list)
-        self.assertEqual(len(arff_tracecontent), 0)
+        self.assertIsNone(trace)
+
+        self._check_fold_timing_evaluations(fold_evaluations, num_repeats, num_folds,
```

Comment:
since neither `num_repeats` nor `num_folds` seem to be used throughout the test other than here, I would prefer the call to just be
```
self._check_fold_timing_evaluations(fold_evaluations, 
    num_repeats=1, num_folds=1,
    task_type=task.task_type_id, check_scores=False)
```
which makes that clear.

Reply:
Done.